mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -65,9 +70,27 @@ def _get_region_name(self):
             s3_endpoint_url=self._s3_endpoint_url,
         )
         try:
-            return temp_client.head_bucket(Bucket=self.bucket)["BucketRegion"]
+            head_bucket_resp = temp_client.head_bucket(Bucket=self.bucket)
+            if _BUCKET_REGION in head_bucket_resp:
+                return head_bucket_resp[_BUCKET_REGION]
+            if (
+                _RESPONSE_METADATA in head_bucket_resp
+                and _HTTP_HEADERS in head_bucket_resp[_RESPONSE_METADATA]
+                and _HTTP_HEADER_BUCKET_REGION
+                in head_bucket_resp[_RESPONSE_METADATA][_HTTP_HEADERS]
+            ):
+                return head_bucket_resp[_RESPONSE_METADATA][_HTTP_HEADERS][
+                    _HTTP_HEADER_BUCKET_REGION
+                ]
+            if _BUCKET_LOCATION_NAME in head_bucket_resp:
+                _logger.warning(
+                    f"Directory bucket {self.bucket} found with BucketLocationName "
+                    f"{head_bucket_resp[_BUCKET_LOCATION_NAME]}."
+                )
+            raise Exception(f"Unable to get the region name for bucket {self.bucket}.")
         except ClientError as error:
-            return error.response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"]
+            _logger.info("Client error caught and getting region from the headers.")
```

Comment:
dont fully understand this block, but should this be logger.warn or logger.error instead?

Reply:
It's not really an error.  Also, I should remove this....this was part of the extra logging for validation that I had added.